libvast/vast/table_slice.hpp

```diff
@@ -40,11 +40,18 @@ class table_slice : public caf::ref_counted {
 
   // -- constructors, destructors, and assignment operators --------------------
 
-  ~table_slice();
+  ~table_slice() override;
+
+  table_slice(const table_slice&) = default;
 
   /// Constructs a table slice with a specific layout.
   /// @param layout The record describing the table columns.
-  table_slice(record_type layout);
+  explicit table_slice(record_type layout);
+
+  // -- factory functions ------------------------------------------------------
+
+  /// Makes a copy of this slice.
+  virtual table_slice_ptr copy() const = 0;
```

Comment:
How about calling this one `clone()`? Then it would be consistent with `abstract_type`.